src/main/java/cloud/eppo/BaseEppoClient.java

```diff
@@ -68,7 +76,10 @@ protected BaseEppoClient(
       BanditLogger banditLogger,
       boolean isGracefulMode,
       boolean expectObfuscatedConfig,
-      Configuration initialConfiguration) {
+      EppoHttpClient httpClient,
```

Comment:
I know this is something we haven't done at Eppo, but I'd like us to start using `@Nullable` and `@NotNull` annotations to indicate which parameters are optional and which ones arent, to help with readability and also IDEs like IntelliJ can use that information to assist with finding bugs. Given `httpClient` is optional, can you please add a `@Nullable` annotation here? Same for `apiKey`, `configStoreOverride` and any other optional params here. It's fine to do incrementally/handle in a separate PR if you prefer